woocommerce/woocommerce-admin
# Category Lookup Table - Fix Category Segments

Closes #1915 

Second attempt at https://github.com/woocommerce/woocommerce-admin/pull/2247 with a different lookup table structure based on @joshuatf suggestion. I'll leave https://github.com/woocommerce/woocommerce-admin/pull/2247 open until this is reviewed.

Creates a category lookup table which stores term relationships for the full category hierarchy.

The table stores data about category hierarchy; the tree and all IDs within. Storing `depth` was unfortunately not straightforward and requires lots of PHP side processing to work out depth per term. I removed that after bashing my head against the desk a few hours.

Data is synced when a category is edited (and changes), and also on install.

Stats work by joining by category_id and then grouping by the tree_id. This join appears to solve our issues in #1915.

To test:

- Update, deactivate and reactivate. Cron job will build the lookup table in short time.
- View Analytics > Categories. Check the totals look good. 
- Try example tests cases which were broken in https://github.com/woocommerce/woocommerce-admin/issues/1915. Totals should include totals from children.
